### PR TITLE
Update recipes to GTest version >=1.13.0

### PR DESCRIPTION
**Pull request by bdice**
This PR updates GTest pinnings to >=1.13.0. This aligns with recent changes in rapids-cmake: https://github.com/rapidsai/rapids-cmake/pull/401.

### CODE REVIEW

File: dependencies.yaml [hunk 1]
```diff
@@ -216,6 +218,12 @@ dependencies:
               - *libcusolver114
               - *libcusparse_dev114
               - *libcusparse114
+  test_libraft:
+    common:
+      - output_types: [conda]
+        packages:
+          - gtest>=1.13.0.*
+          - gmock>=1.13.0.*
```
Comment (bdice) [inline dependencies.yaml:226]:
```suggestion
          - gtest>=1.13.0
          - gmock>=1.13.0
```

**Review by bdice — COMMENTED** [review]
Removing the `.*` to align with changes being made elsewhere (including in cudf where this was likely copied from).

File: conda/environments/all_cuda-118_arch-x86_64.yaml [hunk 2]
```diff
@@ -24,7 +24,9 @@ dependencies:
 - distributed==2023.3.2.1
 - doxygen>=1.8.20
 - gcc_linux-64=11.*
+- gmock>=1.13.0.*
 - graphviz
+- gtest>=1.13.0.*
```
Comment (bdice) [inline conda/environments/all_cuda-118_arch-x86_64.yaml:29]:
```suggestion
- gmock>=1.13.0
- graphviz
- gtest>=1.13.0
```

**Review by cjnolet — APPROVED** [review]

**Comment by bdice** [comment]
I think CI failed due to a flaky test. I lack write permissions on this repo to rerun that single job, so I am merging the upstream to rerun CI.

```
>       assert index.size >= n_rows
E       assert 9969 >= 10000
E        +  where 9969 = Index(type=IVF-FLAT, metric=sqeuclidean, size=9969, dim=10, n_lists=100, adaptive_centers=False).size

python/pylibraft/pylibraft/test/test_ivf_flat.py:132: AssertionError
=========================== short test summary info ============================
FAILED python/pylibraft/pylibraft/test/test_ivf_flat.py::test_extend[device-int8] - assert 9969 >= 10000
 +  where 9969 = Index(type=IVF-FLAT, metric=sqeuclidean, size=9969, dim=10, n_lists=100, adaptive_centers=False).size
======= 1 failed, 1471 passed, 24 skipped, 4 xfailed in 62.83s (0:01:02) =======
```

**Comment by bdice** [comment]
@divyegala You're the best! Thank you so much for your help on this. You're a force multiplier! 💪

**Pull request merged**